### ISSUE TITLE
[enchancement] Common CI script

### ISSUE THREAD

**Issue by atb00ker** [openwisp/openwisp-utils#39]
We need a common CI script to run all the CI tests, so that if there is minor change or a new check is added/removed, we do not need to update all the repositories with the minor change.

Read more: https://github.com/openwisp/openwisp-utils/pull/37#issuecomment-461733269

**Comment by dwang** [comment]
Are these all the scripts that should be run?
- `checkcommit`
- `runcheckendline`
- `runcheckmigration`
- `runflake8`
- `runisort`

However, some of these checks don't apply to all of the repositories, how would you solve that?
For example, `runcheckmigration` wouldn't really apply in `openwisp-website`, but `checkcommit` and `runcheckendline` should still be run.

**Comment by atb00ker** [comment]
@dwang I am thinking command line variable, by default, all tasks will run,
However, if i want to avoid `runcheckmigration`, i can simply do `--no-runcheckmigration` to avoid running that script, what do you think about this? :smile:

**Comment by dwang** [comment]
Sounds good to me!

**Comment by ankit-kumar-dwivedi** [comment]
@atb00ker this sounds great! are you working on this one?

**Comment by atb00ker** [comment]
No. I am currently not working on it. :smile:

**Comment by ankit-kumar-dwivedi** [comment]
okay I am working on it now

**Comment by ankit-kumar-dwivedi** [comment]
@atb00ker in which order should the scripts run? 
I see `checkcommit` and `runcheckendline` should run on every repo.
So I am going to take a maximum of 3 arguments

**Comment by atb00ker** [comment]
Tests are not dependant on each other so the order doesn't matter.
You can checkout the `.travis.yml` file in each repository to checkout what tests are running on that specific repository.
Checkout the same for the following repositories:
```
- django-ipam
- openwisp-ipam
- django-freeradius
- openwisp-freeradius
- openwisp-users
- openwisp-utils
- openwisp-config
- openwisp-controller
- django-loci
- django-netjsonconfig
- django-x509
```

**Comment by ankit-kumar-dwivedi** [comment]
@atb00ker I see that the .travis.yml file in each repository is divided in 3 parts
- **before_install:** here `./runflake8` , `./runisort` , `./runcheckmigration` etc are called
- **install:** here are all install commands...
- **script:** `checkcommit` runs here and reposoitory specific test i.e jshint, ./runtest , `coverage` etc are called.
### Also there are different commands in each .travis.yml to install openwisp-utils
 - `pip install openwisp-utils[qa]>=0.2.1` in openwisp-config
 - `pip install https://github.com/openwisp/openwisp-utils/tarball/master` in OpenWISP-website
 - `pip install https://github.com/openwisp/openwisp-utils/tarball/master#egg=openwisp_utils --upgrade` in openwisp-users
 - `pip install -U https://github.com/openwisp/openwisp-utils/tarball/master` in django-freeradius
 - `pip install --user https://github.com/openwisp/openwisp-utils/tarball/master` in ansible-openwisp

some have the line in requirements-test.txt also a copy of test files present in the repository I am thinking to include the **install command for openwisp-utils** 

**few concerns:**

1. Will it be okay to run the test in script section only? and not before_install (see https://github.com/openwisp/django-ipam/blob/master/.travis.yml)
2. I take the arguments as **--no testname1 testname2 testname2 .....** is it fine?

please let me know your views on this :)

**Comment by unsuitable001** [comment]
@ankit-kumar-dwivedi do not run "before_install" scripts in the "scripts" section. Here's I'm trying to explain why (correction appreciated) -

Running some scripts maybe needed to run the main scripts correctly.
& If any "before_script" or "after_script" fails, it doesn't affects the overall build "status".

**Comment by ppabcd** [comment]
"For example, runcheckmigration wouldn't really apply in openwisp-website, but checkcommit and runcheckendline should still be run."
I think we need add variable for exclude repository. @dwang

**Comment by atb00ker** [comment]
1. Yes, CI tests can run after other tests as well, so as long as it works, i don't see a problem. :smile: 
2. Yep, looks Good. If something is ready, can you please make a [Work in progress] pull request so that we can have a look while you are working on it? 
Thanks. :smile:

**Comment by atb00ker** [comment]
Closed by #45
